controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -2172,16 +2172,18 @@ boolean compareWriterPositions(Map<Long, Long> position1, Map<Long, Long> positi
                               TransactionMetrics.getInstance().reportTransactionBatchCreate(timer2.getElapsed());
                               Timer timer3 = new Timer();
                               return txnCommitOrderer.removeEntities(getScope(), getName(), toPurge)
-                                      .thenAccept(k -> TransactionMetrics.getInstance().reportTransactionZkOrdererPurgeStale(timer3.getElapsed()));
-                          })
-                                        .thenApply(v -> transactionsMap.stream().sorted(
+                                      .thenAccept(k -> {
```

Comment:
this should be thenCompose otherwise that future is dangling

Reply:
fixed